wolfcrypt/src/port/Espressif/esp32_util.c

```diff
@@ -324,12 +343,138 @@ int ShowExtendedSystemInfo(void)
+            words_to_show = 1;
+        }
+    #ifdef MAX_WORDS_ESP_SHOW_MP
+        /* if too big, we'll show MAX_WORDS_ESP_SHOW_MP words */
+        if (words_to_show > MAX_WORDS_ESP_SHOW_MP) {
+            ESP_LOGI(MP_TAG, "Limiting word count from %d to %d",
+                             words_to_show,
+                             MAX_WORDS_ESP_SHOW_MP);
+            words_to_show = MAX_WORDS_ESP_SHOW_MP;
+        }
+    #endif
+        ESP_LOGI(MP_TAG, "%s:",c);
+        esp_show_mp_attributes(c, X);
+        for (size_t i = 0; i < words_to_show; i++) {
+            ESP_LOGI(MP_TAG, "%s.dp[%2d] = 0x%08x;  /* %2d */ ",
+                                   c, /* the supplied variable name      */
+                                   i, /* the index, i for dp[%d]         */
+                                   (unsigned int)X->dp[i], /* the value  */
+                                   i  /* the index, again, for comment   */
+                     );
+        }
+        ESP_LOGI(MP_TAG, "");
+    }
+    return ret;
+}
+
+/* Perform a full mp_cmp and binary compare.
+ * (typically only used during debugging) */
+int esp_mp_cmp(char* name_A, MATH_INT_T* A, char* name_B, MATH_INT_T* B)
+{
+    int ret = MP_OKAY;
+    int e = memcmp(A, B, sizeof(mp_int));
+    if (mp_cmp(A, B) == MP_EQ) {
+        if (e == 0) {
+            /* we always want to be here: both esp_show_mp and binary equal! */
+            ESP_LOGV(TAG, "fp_cmp and memcmp match for %s and %s!",
+                           name_A, name_B);
+        }
+        else {
+            ret = MP_VAL;
+            ESP_LOGE(TAG, "fp_cmp match, memcmp mismatch for %s and %s!",
+                           name_A, name_B);
+            if (A->dp[0] == 1) {
+                ESP_LOGE(TAG, "Both memcmp and fp_cmp fail for %s and %s!",
+                               name_A, name_B);
+            }
+        }
+    }
+    else {
+        ret = MP_VAL;
+        if (e == 0) {
+            /* if mp_cmp says different,
+             * but memcmp says equal, that's a problem */
+            ESP_LOGE(TAG, "memcmp error for %s and %s!",
+                          name_A, name_B);
+        }
+        else {
+            /* in the normal case where mp_cmp and memcmp say the
+             * values are different, we'll optionally show details. */
+            ESP_LOGI(TAG, "e = %d", e);
+            ESP_LOGE(TAG, "fp_cmp mismatch! memcmp "
+                          "offset 0x%02x for %s vs %s!",
+                           e, name_A, name_B);
+            if (A->dp[0] == 1) {
+                ESP_LOGE(TAG, "Both memcmp and fp_cmp fail for %s and %s!",
+                               name_A, name_B);
+            }
+        }
+        ESP_LOGV(TAG, "Mismatch for %s and %s!",
+                       name_A, name_B);
+    }
+
+    if (ret == MP_OKAY) {
+        ESP_LOGV(TAG, "esp_mp_cmp equal for %s and %s!",
+                       name_A, name_B);
+    }
+    else {
+      //  esp_show_mp(name_A, A);
```

Comment:
Remove or use `#if 0` or C style comment `/* */`

Reply:
wrapped in `DEBUG_WOLFSSL`.  This condition is unlikely to occur unless there's a math problem;  will be handy for future HW work, e.g. RISC-V.